src/devices/bus/ieee488/hp9122c.cpp

```diff
@@ -0,0 +1,363 @@
+}
+
+void hp9122c_device::ieee488_ndac(int state)
+{
+	m_i8291a->ndac_w(state);
+}
+
+void hp9122c_device::ieee488_ifc(int state)
+{
+	m_i8291a->ifc_w(state);
+}
+
+void hp9122c_device::ieee488_srq(int state)
+{
+	m_i8291a->srq_w(state);
+}
+
+void hp9122c_device::ieee488_atn(int state)
+{
+	m_i8291a->atn_w(state);
+}
+
+void hp9122c_device::ieee488_ren(int state)
+{
+	m_i8291a->ren_w(state);
+}
+
+WRITE_LINE_MEMBER(hp9122c_device::i8291a_eoi_w)
+{
+	m_bus->eoi_w(this , state);
+}
+
+WRITE_LINE_MEMBER(hp9122c_device::i8291a_dav_w)
+{
+	m_bus->dav_w(this , state);
+}
+
+WRITE_LINE_MEMBER(hp9122c_device::i8291a_nrfd_w)
+{
+	m_bus->nrfd_w(this , state);
+}
+
+WRITE_LINE_MEMBER(hp9122c_device::i8291a_ndac_w)
+{
+	m_bus->ndac_w(this , state);
+}
+
+WRITE_LINE_MEMBER(hp9122c_device::i8291a_ifc_w)
+{
+	m_bus->ifc_w(this , state);
+}
+
+WRITE_LINE_MEMBER(hp9122c_device::i8291a_srq_w)
+{
+	m_bus->srq_w(this , state);
+}
+
+WRITE_LINE_MEMBER(hp9122c_device::i8291a_atn_w)
+{
+	m_bus->atn_w(this , state);
+}
+
+WRITE_LINE_MEMBER(hp9122c_device::i8291a_ren_w)
+{
+	m_bus->ren_w(this , state);
+}
+
+WRITE_LINE_MEMBER(hp9122c_device::i8291a_int_w)
+{
+	m_i8291a_irq = state;
+	update_intsel();
+}
+
+WRITE_LINE_MEMBER(hp9122c_device::i8291a_dreq_w)
+{
+	m_i8291a_drq = state;
+	update_intsel();
+}
+
+WRITE_LINE_MEMBER(hp9122c_device::fdc_intrq_w)
+{
+	m_fdc_irq = state;
+	update_intsel();
+}
+
+WRITE_LINE_MEMBER(hp9122c_device::fdc_drq_w)
+{
+	m_fdc_drq = state;
+	update_intsel();
+}
+
+void hp9122c_device::update_intsel()
+{
+	bool irq = false, firq = false;
+
+	switch(m_intsel & 3) {
+	case 0:
+		/* fdc int connected */
+		irq = m_fdc_drq;
+		firq = m_fdc_irq;
+		break;
+	case 1:
+		/* i8291a ints connected */
+		irq = m_i8291a_drq;
+		firq = m_i8291a_irq;
+		break;
+	case 2:
+		irq = m_index_int;
+		firq = false;
+		break;
+
+	case 3:
+		irq = false;
+		firq = false;
+		break;
+	}
+
+	if (m_cpufirq != firq)
+		m_cpu->set_input_line(M6809_FIRQ_LINE, firq ? ASSERT_LINE : CLEAR_LINE);
+
+	if (m_cpuirq != irq)
+		m_cpu->set_input_line(M6809_IRQ_LINE, irq ? ASSERT_LINE : CLEAR_LINE);
+
+	m_cpuirq = irq;
+	m_cpufirq = firq;
+}
+
+READ8_MEMBER(hp9122c_device::i8291a_dio_r)
+{
+	return m_bus->read_dio();
+}
+
+WRITE8_MEMBER(hp9122c_device::i8291a_dio_w)
+{
+	m_bus->dio_w(this , data);
+}
+
+READ8_MEMBER(hp9122c_device::status_r)
+{
+	uint8_t ret = REG_STATUS_DUAL|REG_STATUS_DISKCHG;
+	auto addr = m_hpib_addr->read();
+
+	if (!m_testmode->read())
+		ret |= 0x10;
+
+	ret |= ((addr & 0x08) << 3) | (addr & 0x07);
+
+	floppy_image_device *floppy = nullptr;
+	if (m_ds0)
+		floppy = m_floppy[0]->get_device();
+	if (m_ds1)
+		floppy = m_floppy[1]->get_device();
+
+	if (floppy) {
+		auto variant = floppy->get_variant();
+		if (variant == floppy_image::DSDD || variant == floppy_image::SSDD)
+			ret |= REG_STATUS_LOW_DENSITY;
+	}
+
+	return ret;
+}
+
+WRITE8_MEMBER(hp9122c_device::cmd_w)
+{
+	auto floppy0 = m_floppy[0]->get_device();
+	auto floppy1 = m_floppy[1]->get_device();
```

Comment:
Please don't use `auto` where the type is:
* Not dependent on template arguments
* Not horribly wrong to write out
* Not a proxy/iterator/something else weird

In cases like this, it doesn't make the cod easier to understand - it just makes it more opaque.